crates/chain_primitives/src/lib.rs

```diff
@@ -26,7 +26,7 @@ pub fn format_token_id(chain: Chain, token_id: String) -> Option<String> {
         | Chain::Ink
         | Chain::Unichain
         | Chain::Hyperliquid
-        | Chain::Monad => Some(EthereumAddress::parse(&token_id)?.to_checksum()),
+        | Chain::Monad => normalize_checksum(&token_id).ok(),
```

Comment:
```suggestion
        | Chain::Monad => ethereum_address_checksum(&token_id).ok(),
```

or just `ethereum_address()`